### PR TITLE
Nuke zips

### PR DESCRIPTION
**Pull request by prusnak**
This PR:
- nukes all ZIPs from the repository
- updates extension.json to use archives from GitHub
- adds more sanity checks to check.py
- one can now run `python3 check.py foo` to check only the `foo` extension and not all of them

### CODE REVIEW

**Comment by prusnak** [comment]
Please pay special attention to extensions `gerty jukebox usermanager` where the hash had to be changed (meaning that the ZIP uploaded earlier was not the same as the generated ZIP by GitHub).

**Review by dni — APPROVED** [review]
dude :D i wanted to make those changes, joking, genius change! LGTM

File: check.py [hunk 1]
```diff
@@ -34,19 +34,25 @@ def __init__(self, ext):
     def validate(self):
         print(f"Checking '{self.name}' extension ({self.id} {self.version})")
 
-        # check for typos
+        # sanity checks
         if not self.name[0].isupper():
-            print(f"{self.name}: name does not start with uppercase letter")
+            print(f"- ERROR: name does not start with uppercase letter")
```
Comment (motorina0) [inline check.py:39]: nitpick: format not required anymore (`f"..."` -> `"..."`)

**Comment by callebtc** [comment]
https://media.tenor.com/KtxkWfiztPwAAAAC/you-dropped-this-king-you-dropped-this-queen.gif

**Review by callebtc — APPROVED** [review]

File: check.py (hunk 1 above)
Comment (callebtc) [inline check.py:39]:
```suggestion
            print("- ERROR: name does not start with uppercase letter")
```

File: check.py [hunk 2]
```diff
@@ -34,19 +34,25 @@ def __init__(self, ext):
...
             assert False
-
         if not self.short_description[0].isupper():
-            print(f"{self.name}: short_description does not start with uppercase letter")
+            print(f"- ERROR: short_description does not start with uppercase letter")
```
Comment (callebtc) [inline check.py:42]:
```suggestion
            print("- ERROR: short_description does not start with uppercase letter")
```

File: check.py [hunk 3]
```diff
@@ -34,19 +34,25 @@ def __init__(self, ext):
...
         if not self.id == self.id.lower():
-            print(f"{self.name}: id has mixed casing")
+            print(f"- ERROR: id has mixed casing")
+            assert False
+        if not self.archive.startswith(self.repo):
+            print(f"- ERROR: archive URL does not start with repo URL")
+            assert False
+        if not self.archive.endswith(f"{self.version}.zip"):
+            bn = basename(self.archive)
+            print(f"- ERROR: archive '{bn}' does not end with version (expected: {self.version}.zip)")
```
Comment (callebtc) [inline check.py:55]:
```suggestion
            print("- ERROR: id has mixed casing")
            assert False
        if not self.archive.startswith(self.repo):
            print("- ERROR: archive URL does not start with repo URL")
            assert False
        if not self.archive.endswith(f"{self.version}.zip"):
            bn = basename(self.archive)
            print(f"- ERROR: archive '{bn}' does not end with version (expected: {self.version}.zip)")
```

**Pull request merged**